tasks/curve-fi-test.ts

```diff
@@ -0,0 +1,183 @@
+    configuration: '0x7974d891822709cf8B1fCB2891AfA9d1BD836D19',
+    depayRouterv1: '0x82154Ea9C2DC4C06D6719cE08728F5cFC9422B1D'
+  }
+}
+
+task('curvefi:test', 'Deploy CurveFi on given network').setAction(
+  async (_args: any, hre: HardhatRuntimeEnvironment) => {
+    // If network is ropsten we will load deployed contracts
+    const db = new JsonDb(hre.network.name === 'ropsten')
+    const [owner] = await hre.ethers.getSigners()
+    const tokenName = ['tokenA', 'tokenB', 'tokenC']
+
+    async function contractDeploy(contractName: string, ...params: any[]) {
+      const instanceFactory =
+        tokenName.indexOf(contractName) >= 0
+          ? await hre.ethers.getContractFactory('TestToken')
+          : await hre.ethers.getContractFactory(contractName)
+      const instance = await instanceFactory.connect(owner).deploy(...params)
+      db.set(contractName, instance.address)
+      console.log('Deploy', contractName, 'at', instance.address)
+      return instance
+    }
+
+    async function contractGet(contractName: string) {
+      const instanceFactory =
+        tokenName.indexOf(contractName) >= 0
+          ? await hre.ethers.getContractFactory('TestToken')
+          : await hre.ethers.getContractFactory(contractName)
+      const instance = instanceFactory.attach(db.get(contractName))
+      console.log('Load', contractName, 'at', instance.address)
+      return instance
+    }
+
+    async function deployIfNotExist(contractName: string, ...params: any[]) {
+      return !db.has(contractName) ? contractDeploy(contractName, ...params) : contractGet(contractName)
+    }
+
+    // Deploy needed contracts of CurveFi
+    const instanceAddressProvider = <AddressProvider>await deployIfNotExist('AddressProvider', owner.address)
+    const instanceCalculator = <CurveCalc>await deployIfNotExist('CurveCalc')
+    const instanceGaugeControllerMock = <GaugeControllerMock>await deployIfNotExist('GaugeControllerMock')
+    const instancePoolInfo = <PoolInfo>await deployIfNotExist('PoolInfo', instanceAddressProvider.address)
+    const instanceRegistry = <Registry>(
+      await deployIfNotExist('Registry', instanceAddressProvider.address, instanceGaugeControllerMock.address)
+    )
+    // Add registry to address provider
+    await instanceAddressProvider.connect(owner).add_new_id(instanceRegistry.address, 'CurveFi Registry')
+    await instanceAddressProvider.connect(owner).set_address(0, instanceRegistry.address)
+    const instanceSwaps = <Swaps>(
+      await deployIfNotExist('Swaps', instanceAddressProvider.address, instanceCalculator.address)
+    )
+    const instanceLpToken = <LpToken>await deployIfNotExist('LpToken', 'CurveFi', 'CRV', 18, 0)
+
+    // Deploy test token
+    const tokenA = await deployIfNotExist('tokenA')
+    const tokenB = await deployIfNotExist('tokenB')
+    const tokenC = await deployIfNotExist('tokenC')
+    const tokenSETH = await deployIfNotExist('WETH9')
```

Comment:
Are you sure this shouldn't be `deployIfNotExist('SETH')` ? as WETH!=SETH 🤔

Reply:
I will try to add SETH for testing.